dot/state/pruner.go

```diff
@@ -0,0 +1,363 @@
+package state
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ChainSafe/chaindb"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/scale"
+)
+
+const (
+	journalPrefix = "journal"
+	lastPrunedKey = "last_pruned"
+	pruneInterval = time.Second
+)
+
+// nolint
+const (
+	FullNode    = "full"
+	ArchiveNode = "archive"
+)
+
+// Pruning online pruning mode of historical state tries
+type Pruning string
+
+// IsValid checks whether the pruning mode is valid
+func (p Pruning) IsValid() bool {
+	switch p {
+	case FullNode:
+		return true
+	case ArchiveNode:
+		return true
+	default:
+		return false
+	}
+}
+
+// String returns format of Pruning
+func (p Pruning) String() string {
+	switch p {
+	case FullNode:
+		return "full"
+	case ArchiveNode:
+		return "archive"
+	default:
+		return ""
+	}
+}
+
+// Pruner is implemented by fullNodePruner and archivalNodePruner.
+type Pruner interface {
```

Comment:
no public methods? does this need to be public then?

Reply:
Updated